common/zeroconf/target.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+	Aseba - an event-based framework for distributed robot control
+	Copyright (C) 2007--2016:
+		Stephane Magnenat <stephane at magnenat dot net>
+		(http://stephane.magnenat.net)
+		and other contributors, see authors.txt for details
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, version 3 of the License.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../utils/utils.h"
+#include "../utils/FormatableString.h"
+#include <dashel/dashel.h>
+#include "zeroconf.h"
+
+using namespace std;
+
+namespace Aseba
+{
+	//! This target is described by a human-readable name and a port.
+	//! If a port is nonzero at creation then this is a local target to be remembered
+	//! otherwise it is a remote one discovered through browsing and may be refreshed.
+	//! Note that later the port will become nonzero if the target is resolved.
+	Zeroconf::Target::Target(const std::string & name, const int & port) :
```

Comment:
It is a detail but I do not see why a int (4 bytes) should be passed as reference (8 bytes)